crates/ruff_workspace/src/options.rs

```diff
@@ -1878,6 +1878,118 @@ pub struct Flake8TypeCheckingOptions {
         "#
     )]
     pub quote_annotations: Option<bool>,
+
+    /// Whether to add quotes around the type expression of a `typing.cast`
+    /// call, if doing so would allow the corresponding import to be moved
+    /// into a type-checking block.
+    ///
+    /// For example, in the following, Python requires that `Sequence` be
+    /// available at runtime, despite the fact that it's only used in a `cast`
+    /// which doesn't do anything at runtime.
+    ///
+    /// ```python
+    /// from collections.abc import Sequence
+    /// from typing import cast
+    ///
+    /// cast(Sequence[float], [1, 2, 3])
+    /// ```
+    ///
+    /// In other words, moving `from collections.abc import Sequence` into an
+    /// `if TYPE_CHECKING:` block above would cause a runtime error, as the
+    /// type would no longer be available at runtime.
+    ///
+    /// By default, Ruff will respect such runtime semantics and avoid moving
+    /// the import to prevent such runtime errors.
+    ///
+    /// Setting `quote-cast-type-expressions` to `true` will instruct Ruff to
+    /// add quotes around the type expression (e.g., `"Sequence[float]"`),
+    /// which in turn enables Ruff to move the import into an
+    /// `if TYPE_CHECKING:` block, like so:
+    ///
+    /// ```python
+    /// from typing import TYPE_CHECKING, cast
+    ///
+    /// if TYPE_CHECKING:
+    ///     from collections.abc import Sequence
+    ///
+    /// cast("Sequence[float]", [1, 2, 3])
+    /// ```
+    ///
+    /// Note that this setting should always be safe to enable, but can lead
+    /// to inconsistent style with `typing.cast` calls where sometimes the
+    /// type expression is quoted or partially quoted and other times not,
+    /// if you'd like to consistently quote type expressions, you should
+    /// instead consider enabling [ruff_linter::registry::Rule::RuntimeCastValue].
```

Comment:
I'm guessing this is not the right way to cross reference another rule, do I have to manually link to the rule's page? It seems like all the other references to other settings in here manually link via fragments, rather than relying on the docs to create the correct cross reference.